Automattic/woocommerce-subscriptions-core
# [HPOS] Refactor `WCS_Meta_Box_Schedule::save` to fix PHP notices that occur when updating an order via the Edit Order page

Partially fixes #196

This branch is based on #280.

## Description







This PR fixes PHP notices related to `WCS_Meta_Box_Schedule::save` that occur when updating an order via the Edit Order page with HPOS enabled.

## How to test this PR





0. [Enable HPOS](https://github.com/woocommerce/woocommerce/wiki/High-Performance-Order-Storage-Upgrade-Recipe-Book#how-to-enable-hpos).
2. Purchase a Subscription product, noting the order ID.
3. Go to WooCommerce → Orders and navigate to the order's Edit Order page.
4. Select a new order **status** (processing/pending payment) and click update.
5. The order should update successfully and you should not see PHP notices related to **`WCS_Meta_Box_Schedule::save`**.
		(There may be others, e.g. `WCS_Meta_Box_Subscription_Data::save`).
3. Go to WooCommerce → Subscriptions and navigate to a subscription's Edit Subscription page.
4. Select a new subscription status (on hold/active) and click update.
5. The subscription should update successfully and you should not see PHP notices related to **`WCS_Meta_Box_Schedule::save`**.
4. Change the subscription billing period and interval and click update. The change should be reflected.
6. Disable HPOS and repeat steps 2-8
	(WC Settings → Advanced → Custom data stores →  Use the WordPress posts table)

**Note**: you may see the following error when setting a Subscription status to `on-hold` with HPOS enabled, syncing disabled. A new issue has been created for this: #293

```
Invalid data. First parameter was empty when passed to update_dates().
```

## Product impact
